src/OnlineVotingS.Application/Services/ImplService/CampaignService.cs

```diff
@@ -12,71 +13,176 @@
+    {
+        _logger.LogInformation("Updating campaign with ID {CampaignId}.", campaignDto.CampaignID);
 
-        public async Task<Campaign> UpdateCampaignAsync(CampaignPutDTO campaignDto)
+        try
         {
             var campaign = await _campaignRepository.GetByIdAsync(campaignDto.CampaignID);
             if (campaign == null)
             {
+                _logger.LogWarning("Campaign with ID {CampaignId} not found.", campaignDto.CampaignID);
                 throw new KeyNotFoundException($"Campaign with ID {campaignDto.CampaignID} not found.");
             }
 
             _mapper.Map(campaignDto, campaign);
             await _campaignRepository.UpdateAsync(campaign);
+
+            _logger.LogInformation("Campaign with ID {CampaignId} updated successfully.", campaignDto.CampaignID);
             return campaign;
         }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "An error occurred while updating the campaign with ID {CampaignId}.", campaignDto.CampaignID);
+            throw;
+        }
+    }
+
+    public async Task<bool> DeleteCampaignAsync(int campaignId)
+    {
+        _logger.LogInformation("Deleting campaign with ID {CampaignId}.", campaignId);
 
-        public async Task<bool> DeleteCampaignAsync(int campaignId)
+        try
         {
             var exists = await _campaignRepository.ExistsAsync(campaignId);
             if (!exists)
             {
+                _logger.LogWarning("Campaign with ID {CampaignId} not found.", campaignId);
                 throw new KeyNotFoundException($"Campaign with ID {campaignId} not found.");
             }
 
             await _campaignRepository.DeleteAsync(campaignId);
+
+            _logger.LogInformation("Campaign with ID {CampaignId} deleted successfully.", campaignId);
             return true;
         }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "An error occurred while deleting the campaign with ID {CampaignId}.", campaignId);
+            throw;
+        }
+    }
 
-        public async Task<Campaign> GetCampaignByIdAsync(int campaignId)
+    public async Task<Campaign> GetCampaignByIdAsync(int campaignId)
+    {
+        _logger.LogInformation("Fetching campaign with ID {CampaignId}.", campaignId);
+
+        try
         {
-            return await _campaignRepository.GetByIdAsync(campaignId);
+            var campaign = await _campaignRepository.GetByIdAsync(campaignId);
+            if (campaign == null)
+            {
+                _logger.LogWarning("Campaign with ID {CampaignId} not found.", campaignId);
+                throw new KeyNotFoundException($"Campaign with ID {campaignId} not found.");
+            }
+
+            return campaign;
         }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "An error occurred while fetching the campaign with ID {CampaignId}.", campaignId);
```

Comment:
follow please if it works the same guidline as I mentonied here https://github.com/ArianitSGashi/OnlineVotingS/pull/45/files#r1723195909